src/tests/functional/shared_test_classes/src/base/ov_subgraph.cpp

```diff
@@ -307,6 +307,12 @@ void SubgraphBaseTest::compile_model() {
     }
     try {
         inference_precision = core->get_property(targetDevice, ov::hint::inference_precision);
+        // for avx2_vnni_2 platforms use the set inference_precision hint in case generate higher threshold during
+        // following calculate_thresholds stage
+        if (ov::with_cpu_x86_avx2_vnni_2() &&
+            configuration.find(ov::hint::inference_precision.name()) != configuration.end()) {
+            inference_precision = configuration.at(ov::hint::inference_precision.name()).as<ov::element::Type>();
+        }
```

Comment:
@liubo-intel  That shouldn't be part of common logic. 
I bet the issue we face is that inference_precision specified in core and inference_precision specified in compiled_model are different. Is my understanding correct?
@iefode @sbalandi What do you think about updating line 309 to `inference_precision = compiledModel.get_property(ov::hint::inference_precision));` as it reflects real inference precision for given test.

Reply:
That's why I am saying we need to use `compiledModel.get_property(ov::hint::inference_precision));` as it will return bf16/fp16 if the test explicitly forces it

Reply:
> That's why I am saying we need to use `compiledModel.get_property(ov::hint::inference_precision));` as it will return bf16/fp16 if the test explicitly forces it

Hi, @dmitry-gorokhov : yes, this changes can solve LNL platform problems, but this will also change behavior(mainly the threshold) on current platforms(e.g. GNR) which make some previously passed testcase fail now on these platforms. 
do you suggest we limit this changes to LNL+ platform in this pr or we should also cover other platforms now(mainly GNR I think)

Reply:
I actually do not expect new test failures given the fact the threshold will become less restrictive on SPR/GNR. Lets have a try

Reply:
based on our offline sync, have fallback to the previous threshold for some sensitive test cases after we updated the 'inference_precision' method of 'SubgraphTest', which may have made the threshold stricter